src/main/java/randoop/resource/MethodListGen.java

```diff
@@ -0,0 +1,182 @@
+
+      // Randoop expects a list of omitmethods as regex.
+      if (nonDetMethods.size() > 0) {
+        System.out.println("Nondeterministic methods count: " + nonDetMethods.size());
+        try (BufferedWriter nonDetMethodWriter = Files.newBufferedWriter(nonDetFile, UTF_8)) {
+          nonDetMethodWriter.write(OMIT_METHODS_FILE_HEADER);
+          for (String method : nonDetMethods) {
+            nonDetMethodWriter.write("^" + Pattern.quote(method) + "$");
+            nonDetMethodWriter.newLine();
+          }
+        }
+      } else {
+        System.out.println("No nondeterministic methods found. Not writing to file.");
+      }
+
+      List<String> sideEffectFreeMethods =
+          getAnnotatedMethodsFromJar(annotatedJar, SEF_ANNOTATIONS);
+
+      // There are some fully qualified signatures that Randoop cannot parse.
+      // We will separate these into two files, the default (parsable) file and
+      // the non-parsable file.
+      if (sideEffectFreeMethods.size() > 0) {
+        System.out.println("Total side effect free methods count: " + sideEffectFreeMethods.size());
+        try (BufferedWriter sideEffectMethodWriter =
+                Files.newBufferedWriter(sideEffectFreeFile, UTF_8);
+            BufferedWriter unparsableSideEffectMethodWriter =
+                Files.newBufferedWriter(unparsableSideEffectFreeFile, UTF_8)) {
+          for (String fullyQualifiedMethodSignature : sideEffectFreeMethods) {
+            try {
+              signatureToOperation(
+                  fullyQualifiedMethodSignature,
+                  VisibilityPredicate.IS_ANY,
+                  new EverythingAllowedPredicate());
+              sideEffectMethodWriter.write(fullyQualifiedMethodSignature);
+              sideEffectMethodWriter.newLine();
+            } catch (RandoopUsageError | FailedPredicateException e) {
+              System.err.println("Not parsable: " + e.getMessage());
+              unparsableSideEffectMethodWriter.write(fullyQualifiedMethodSignature);
+              unparsableSideEffectMethodWriter.newLine();
+              e.printStackTrace(new PrintWriter(unparsableSideEffectMethodWriter));
+              unparsableSideEffectMethodWriter.newLine();
+            }
+          }
+        }
+      } else {
+        System.out.println("No side effect free methods found. Not writing to file.");
+      }
+    } catch (IOException e) {
+      e.printStackTrace();
+      System.exit(1);
+    }
+  }
+
+  /**
+   * Returns the annotated methods from the the given jar.
+   *
+   * @param jarFile .jar file path
+   * @param annotations which annotations to capture
+   * @return list of annotated methods in fully-qualified signature format, in alphabetical order
+   * @throws IOException if Randoop cannot parse a method
+   */
+  private static List<String> getAnnotatedMethodsFromJar(
+      Path jarFile, Collection<String> annotations) throws IOException {
```

Comment:
`throws IOException` does not seem accurate.  The code catches `IOException`, and code above catches `RandoopUsageError | FailedPredicateException`.